doc/source/regression/dwt-idwt.md

```diff
@@ -0,0 +1,214 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 1.0.0
+    jupytext_version: 1.16.1
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+mystnb:
+  execution_allow_errors: true
+  execution_show_tb: true
+---
+
+(reg-dwt-idwt)=
+
+```{eval-rst}
+.. currentmodule:: pywt
+
+.. dropdown:: 🧑‍🔬 This notebook can be executed online. Click this section to try it out! ✨
```

Comment:
Is this how it's done in scikit-learn as well? I don't mind if it's needed, but it seems easy to forget this kind of snippet in larger projects.

Reply:
It should be possible to write another minimal Sphinx extension that can automate this and connect to the HTML builder at an event, similar to `convert_md_to_ipynb`. We indeed don't require it here, however.

As for scikit-learn, I think they switched to the PyData Sphinx Theme as well in their recent release, and I feel like I like their solution much more than using an admonition – the JupyterLite/Binder/download buttons are very conveniently placed in the secondary sidebar. I can try to explore that and see how they did it, and whether we can do it conditionally for just these six pages through the theme's API.

Reply:
Here's an example: https://scikit-learn.org/stable/auto_examples/classification/plot_classifier_comparison.html#sphx-glr-auto-examples-classification-plot-classifier-comparison-py

Reply:
https://github.com/sphinx-gallery/sphinx-gallery/pull/1312 is a non-minimal implementation, so we can't copy it that easily unless it were a Sphinx extension of its own.

Reply:
That looks really nice! I suggest then that we merge this as is, and explore the option of how to have support either in pydata-sphinx-theme or in a separate Sphinx extension for having a right sidebar like done in sphinx-gallery.

Reply:
Thanks! I'm still figuring out a solution for preprocessing the notebooks to add the directives on-demand (it's not that easy to do without a Markdown parser that allows me to change things programmatically through a document's AST). 


Another option I have explored is an [extension](https://www.ericholscher.com/blog/2016/jul/25/integrating-jinja-rst-sphinx/) that runs at the time of the `source-read` event, and I have it working partially. If this doesn't work, I can ping you to merge this – because the content looks complete and I guess we will have better things to spend more time on.